src/rgw/rgw_sal.h

```diff
@@ -134,7 +134,8 @@ class Store {
 
     /* This one does not query the cluster for info */
     virtual std::unique_ptr<User> get_user(const rgw_user& u) = 0;
-    /* These three do query the cluster for info */
+    /* These do query the cluster for info */
+    virtual std::string get_cluster_id(const DoutPrefixProvider* dpp,  optional_yield y) = 0;
```

Comment:
the idea is, yes, to distinguish between different clusters, e.g., when k8s is interacting with several

I think the idea of zipperifying it would be to ask the store implementation to provide a unique string--it could be an ordinary uuid produced (e.g.) by uuidgen, or anything else the store preferred

Reply:
tangentially, I don't think that linking a standalone store to a ceph cluster is something we should expose as a "thing"--if it is convenient to work in the next phase of zipper, that's obviously fine, but the end goal of zipper is for store backends to be unique;  I'm not sure about different layerings--maybe that can't be unique;

Reply:
broadly, I'm trying to contribute to the goal of main code paths -only- talking to sal, not making special case calls into rados